pkg/container/qemu_runner.go

```diff
@@ -118,7 +118,38 @@ func (bw *qemu) Run(ctx context.Context, cfg *Config, envOverride map[string]str
 }
 
 func (bw *qemu) Debug(ctx context.Context, cfg *Config, envOverride map[string]string, args ...string) error {
-	clog.InfoContextf(ctx, "debugging command %s", strings.Join(args, " "))
+	clog.FromContext(ctx).Debugf("running debug command: %v", args)
+
+	// default to root user, unless a different user is specified
+	user := "root"
+	if cfg.RunAs != "" {
+		user = cfg.RunAs
+	}
+
+	log.Debug("qemu: ssh - get user ssh key pair")
+	pubKey, err := getUserSSHKey()
+	if err != nil {
+		log.Warn("qemu: could not get user ssh key pair, using ephemeral ones")
+	}
+	if pubKey != nil {
+		command := fmt.Sprintf("echo '%s' | tee -a /root/.ssh/authorized_keys /home/*/.ssh/authorized_keys", string(pubKey))
```

Comment:
I think this is "safer":
```suggestion
		command := fmt.Sprintf("echo %q | tee -a /root/.ssh/authorized_keys /home/*/.ssh/authorized_keys", string(pubKey))
```